lib/application.ts

```diff
@@ -783,16 +793,17 @@ export class SNApplication {
       } else {
         return payload;
       }
-    })
-    const affectedUuids = await this.modelManager!.importPayloads(validPayloads);
+    });
+    const validPayloadsUuids = await this.modelManager!.importPayloads(validPayloads);
```

Comment:
If you're eventually going to import both valid and invalid payloads, why even separate them in the first place?

Reply:
Because the order matters. If we import everything at once, some payloads may not be decrypted. This makes sure all payloads are imported correctly. 

Additionally:

https://github.com/standardnotes/snjs/blob/a27979314775f7aa34d4817bee740ab3fa3948c5/lib/application.ts#L769

won't have payloads that are `waitingForKeys`. If `waitingForKeys` then `errorDecrypting` is `true` according to [this](https://github.com/standardnotes/snjs/blob/a27979314775f7aa34d4817bee740ab3fa3948c5/lib/services/protocol_service.ts#L527) line.

Another solution would be to import _validPayloads_ first and then _payloadsWaitingForKeys_. But in the end, we can't import them both at the same time.

Reply:
Why would order matter? [This line](https://github.com/standardnotes/snjs/blob/004/lib/services/protocol_service.ts#L146) would monitor items key insertion/changes and decrypt any errorred items. So even if after calling importData not all items are immediately decrypted, they would _eventually_ be decrypted in another run cycle. But it would seem this problem here anyway would not be solved by importing data separately. Instead, it seems like you would want to modify `protocolService.payloadsByDecryptingBackupFile` to decrypt items keys first, then other item types.